ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -175,45 +147,59 @@ export const MultiSelectWithLabel = (props) => {
   </FlexColumn>;
 };
 
+export interface AccountCreationProps {
+  profile: Profile;
+  invitationKey: string;
+  onComplete: (profile: Profile) => void;
+}
+
+export interface AccountCreationState {
+  creatingAccount: boolean;
+  errors: any;
+  invalidEmail: boolean;
+  rolesOptions: any;
+  profile: Profile;
+  showAllFieldsRequiredError: boolean;
+  showInstitution: boolean;
+  showNonAcademicAffiliationRole: boolean;
+  showNonAcademicAffiliationOther: boolean;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  institutionName: string;
+  institutionRole: string;
+  nonAcademicAffiliation: string;
+  nonAcademicAffiliationRole: string;
+  nonAcademicAffiliationOther: string;
+}
+
 export class AccountCreation extends React.Component<AccountCreationProps, AccountCreationState> {
   private usernameCheckTimeout: NodeJS.Timer;
 
   constructor(props: AccountCreationProps) {
+    if (props.profile.institutionalAffiliations.length !== 1) {
+      throw new Error('Profile must be pre-allocated with 1 institutional affiliation.');
```

Comment:
I don't understand this precondition. Is this for the new account creation logic, or the old?

Reply:
Thanks, this seems fine. I think the only thing that may be confusing here is that typically this profile is going to be passed in an empty state. My first thought on reading this is that perhaps we're expecting the affiliation to already be filled in. A comment here explaining that it will typically be just a placeholder empty affiliation might make this more clear.